apps/passport-server/test/pretix/GenericPretixDataMocker.ts

```diff
@@ -312,55 +371,70 @@ export class GenericPretixDataMocker {
     };
   }
 
-  private newItem(
-    name: string,
-    category: number,
-    addon = false
-  ): GenericPretixItem {
+  /**
+   * E.g. GA, Bouncer, T-Shirt, Towel
+   */
+  private newProductType(
+    productName: string,
+    productCategoryId: number,
+    { is_addon } = { is_addon: false }
+  ): GenericPretixProduct {
     return {
       id: this.nextId(),
-      name: { en: name },
-      category,
-      admission: !addon,
-      personalized: !addon
+      name: { en: productName },
+      category: productCategoryId,
+      admission: !is_addon,
+      personalized: !is_addon
     };
   }
 
-  private newPretixOrder(
+  /**
+   * @param orderEmail - array of (product id, attendee email) tuples,
+   */
+  private newOrder(
     orderEmail: string,
-    itemsAndEmails: [number, string | null][] // array of (item, attendee email) tuples,
+    productEmailPairs: [number, string | null][]
   ): GenericPretixOrder {
     const orderId = this.randomOrderCode();
 
     return {
       code: orderId,
-      name: this.randomName(),
-      status: "p",
+      /**
+       * name attached to order doesn't have to match names
+       * on positions (individual purchased items)
+       */
+      name: randomName(),
+      status: "p", // p = paid
       testmode: false,
-      secret: this.randomSecret(),
+      secret: this.randomPositionSecret(),
       email: orderEmail,
-      positions: itemsAndEmails.map(([item, email]) =>
-        this.newPosition(orderId, email, item, this.nextId())
+      positions: productEmailPairs.map(([product, email]) =>
+        // TODO @rob - is the sub event id meant to be using `this.nextId()`
```

Comment:
I honestly don't know. I'm not even sure what a "sub event" is - I just copied this from the Devconnect mock test generator.